stubs/Werkzeug/werkzeug/datastructures.pyi

```diff
@@ -217,7 +220,10 @@ class ImmutableHeadersMixin:
     add_header: Any
     def extend(self, iterable): ...
     def insert(self, pos, value): ...
-    def pop(self, **kwargs): ...
+    @overload
+    def pop(self, key: str, default: Optional[str] = ...) -> str: ...
+    @overload
+    def pop(self, key: Optional[int], default: Optional[Tuple[str, str]] = ...) -> Tuple[str, str]: ...
```

Comment:
This method seems to be something completely different than the other `pop`. It is defined as:

```
    def pop(self, index=-1):
        is_immutable(self)
```

While  the other pop is like `dict.pop`, accepting `key` and `default`, this is more like `list.pop`.

Reply:
Ok got that. I only added that method as they handled that in their original code [here](https://github.com/pallets/werkzeug/blob/bb8d422329750f36575ba19597240ff798fbb268/src/werkzeug/datastructures.pyi#L246). But, I guess what you said makes more sense.